expression/builtin_control.go

```diff
@@ -67,9 +67,8 @@ func InferType4ControlFuncs(lhs, rhs *types.FieldType) *types.FieldType {
 	} else if rhs.Tp == mysql.TypeNull {
 		*resultFieldType = *lhs
 	} else {
-		var unsignedFlag uint
-		evalType := types.AggregateEvalType([]*types.FieldType{lhs, rhs}, &unsignedFlag)
 		resultFieldType = types.AggFieldType([]*types.FieldType{lhs, rhs})
+		evalType := types.AggregateEvalType([]*types.FieldType{lhs, rhs}, &resultFieldType.Flag)
```

Comment:
Is it safe to directly modify `resultFieldType.Flag` in `AggregateEvalType()`?

Reply:
It should be okay to do it since we already do the same thing when building `coalesce` functions ([in this line](https://github.com/pingcap/tidb/blob/master/expression/builtin_compare.go#L119)) and the second argument of `AggregateEvalType` is a pointer which should be used to adjust the flag from the outside.